common/ipfs/api.go

```diff
@@ -41,18 +43,40 @@ func NewIpfsApiWithCore(logger models.Logger, multiAddressStr string, coreApi if
 	return &ipfs
 }
 
-func NewIpfsApi(logger models.Logger, multiAddressStr string, metricService models.MetricService) *IpfsApi {
-	addr, err := ma.NewMultiaddr(multiAddressStr)
+func parseAddress(addrStr string) (ma.Multiaddr, error) {
+	if addr, err := ma.NewMultiaddr(addrStr); err == nil {
+		return addr, nil
+	}
+
+	u, err := url.Parse(addrStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid address %s given: %w", addrStr, err)
+	}
+
+	host, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		return nil, fmt.Errorf("invalid address %s given: %w", addrStr, err)
+	}
+
+	if addr, err := ma.NewMultiaddr(fmt.Sprintf("/dns4/%s/tcp/%s", host, port)); err != nil {
```

Comment:
This doesn't seem right. We're parsing the passed multiaddr then forcing it to conform to `/dns4/%s/tcp/%s`, which may or may not be valid.